src/realm/mixed.hpp

```diff
@@ -408,17 +408,59 @@ inline bool Mixed::is_null() const
     return (m_type == 0);
 }
 
-namespace {
-inline int comp(StringData a, StringData b)
+namespace _impl {
+inline int compare_string(StringData a, StringData b)
 {
     if (a == b)
         return 0;
     return utf8_compare(a, b) ? -1 : 1;
 }
+
+template <int>
+struct IntTypeForSize;
+template <>
+struct IntTypeForSize<1> {
+    using type = uint8_t;
+};
+template <>
+struct IntTypeForSize<2> {
+    using type = uint16_t;
+};
+template <>
+struct IntTypeForSize<4> {
+    using type = uint32_t;
+};
+template <>
+struct IntTypeForSize<8> {
+    using type = uint64_t;
+};
+
+template <typename Float>
+inline int compare_float(Float a_raw, Float b_raw)
+{
+    bool a_nan = std::isnan(a_raw);
+    bool b_nan = std::isnan(b_raw);
+    if (!a_nan && !b_nan) {
+        // Just compare as IEEE floats
+        return a_raw == b_raw ? 0 : a_raw < b_raw ? -1 : 1;
+    }
+    if (a_nan && b_nan) {
+        // Compare the nan values as unsigned
+        using IntType = typename _impl::IntTypeForSize<sizeof(Float)>::type;
+        IntType a = 0, b = 0;
+        memcpy(&a, &a_raw, sizeof(Float));
```

Comment:
why does this go over memory. Wouldn't it be smarter to just reinterpret_cast<> from float to corresponding inttype?